node/worker/worker.go

```diff
@@ -86,15 +83,36 @@ func (w *Worker) CommitTransactions(
 			continue
 		}
 		// Start executing the transaction
-		w.current.state.Prepare(tx.Hash(), common.Hash{}, len(w.current.txs))
+		w.current.state.Prepare(tx.Hash(), common.Hash{}, dbIndex)
+		dbIndex = dbIndex + 1
```

Comment:
Here the index counter should only increment when the txn succeeded. Failed txns won't be included in the new block.